FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +19,46 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: self.url) { [weak self] (result) in
+			guard self != nil else { return }
+			switch result {
+			case .failure(_):
+				completion(.failure(Error.connectivity))
+			case .success(let (data, httpResponse)):
+				do {
+					let feed = try FeedImageMapper.map(data, httpResponse)
+					completion(.success(feed))
+				} catch {
+					completion(.failure(Error.invalidData))
+				}
+			}
+		}
+	}
+}
+
+private class FeedImageMapper {
+	static private var OK_200 = 200
+
+	private struct Root: Decodable {
+		let items: [Items]
+	}
+
+	private struct Items: Decodable {
+		let image_id: UUID
+		let image_desc: String?
+		let image_loc: String?
+		let image_url: URL
+
+		var feed: FeedImage {
+			return FeedImage(id: image_id, description: image_desc, location: image_loc, url: image_url)
+		}
+	}
+
+	static func map(_ data: Data, _ reponse: HTTPURLResponse) throws -> [FeedImage] {
```

Comment:
There is a typo in the `response` label:

```suggestion
	static func map(_ data: Data, _ response: HTTPURLResponse) throws -> [FeedImage] {
```